### PR TITLE
feat: add Solana alert

### PR DESCRIPTION
**Pull request by dennyscode**
Ticket: https://lifi.atlassian.net/browse/LF-7118

image
image

### CODE REVIEW

File: src/components/SolanaAlert/SolanaAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <InfoMessageCard mt={theme.spacing(4)} mb={theme.spacing(4)}>
+      <InfoMessageCardTitle display="flex" alignItems="center" px={2} pt={2}>
+        <InfoIcon
+          sx={{
+            marginRight: 1,
+          }}
+        />
+        <Typography variant={'lifiHeaderXSmall'}>
+          {t('solanaAlert.title')}
```
Comment (dennyscode) [inline src/components/SolanaAlert/SolanaAlert.tsx:18]: Title: "Solana swaps are currently limited!"

Do wee keep it that way?

File: src/components/SolanaAlert/SolanaAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        </Typography>
+      </InfoMessageCardTitle>
+      <Typography variant={'lifiBodySmall'} pt={theme.spacing(1.5)}>
+        {t('solanaAlert.subtitle')}
```
Comment (dennyscode) [inline src/components/SolanaAlert/SolanaAlert.tsx:22]: Subtitle: "Only bridge from stablecoin to stablecoin are possible when bridging from or to Solana"

Do we keep it that way?

File: src/components/Widgets/Widgets.tsx [hunk 3]
```diff
@@ -100,6 +107,12 @@ export function Widgets() {
       >
         <Widget starterVariant={TabsMap.Refuel.variant as WidgetSubvariant} />
       </WidgetContainer>
+      {(sourceChainToken.chainId === ChainId.SOL ||
+        destinationChainToken.chainId === ChainId.SOL) && (
```
Comment (dennyscode) [inline src/components/Widgets/Widgets.tsx:111]: Check if either Source- or Destination-ChainID matches SOL-ChainID

File: src/i18n/en/translation.json [hunk 4]
```diff
@@ -60,13 +60,17 @@
     "testnet": "Any transactions or actions performed on the Testnet will not be reflected on the Mainnet.",
     "switchToMainnet": "Switch to Mainnet"
   },
+  "solanaAlert": {
+    "title": "Solana swaps are currently limited!",
+    "subtitle": "Only bridge from stablecoin to stablecoin are possible when bridging from or to Solana"
```
Comment (dennyscode) [inline src/i18n/en/translation.json:65]: Do we keep it that way?

File: src/i18n/en/translation.json [hunk 5]
```diff
@@ -60,13 +60,17 @@
     "testnet": "Any transactions or actions performed on the Testnet will not be reflected on the Mainnet.",
     "switchToMainnet": "Switch to Mainnet"
   },
+  "solanaAlert": {
+    "title": "Solana swaps are currently limited!",
```
Comment (dennyscode) [inline src/i18n/en/translation.json:64]: Do wee keep it that way?

File: src/i18n/en/translation.json [hunk 6]
```diff
@@ -60,13 +60,17 @@
...
+    "subtitle": "Only bridge from stablecoin to stablecoin are possible when bridging from or to Solana"
+  },
   "featureCard": {
     "learnMore": "Learn more"
   },
   "button": {
     "okay": "Okay"
   },
-  "error" : {
+  "error": {
```
Comment (dennyscode) [inline src/i18n/en/translation.json:73]: Guess, the linter did its thing here

File: src/stores/chainTokenSelection/chainTokenSelection.tsx
Comment (dennyscode) [inline src/stores/chainTokenSelection/chainTokenSelection.tsx:1]: It seems to be all about stores these days, so why not create a new one for the chain & token selections for source + destination?

File: src/stores/chainTokenSelection/chainTokenSelection.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  setSourceChainToken: (sourceChainToken: ChainTokenSelected) => void;
+  destinationChainToken: OptionalChainTokenSelected;
+  setDestinationChainToken: (destinationChainToken: ChainTokenSelected) => void;
+}
+
+const queryParameters = new URLSearchParams(window.location.search);
+const fromChain = queryParameters.get('fromChain');
+const fromToken = queryParameters.get('fromToken');
+const toChain = queryParameters.get('toChain');
+const toToken = queryParameters.get('toToken');
```
Comment (dennyscode) [inline src/stores/chainTokenSelection/chainTokenSelection.tsx:21]: Also, covering the URL-params to render that Solana alert on page refresh after a route has already been selected.

File: src/stores/chainTokenSelection/chainTokenSelection.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+      destinationChainToken: {
+        tokenAddress: toToken || null,
+        chainId: (toChain && (parseInt(toChain) as ChainId)) || null,
+      },
+      setDestinationChainToken: (destinationChainToken) =>
+        set({
+          destinationChainToken,
+        }),
+    }),
+    Object.is,
```
Comment (dennyscode) [inline src/stores/chainTokenSelection/chainTokenSelection.tsx:43]: No persistent storage, just needed to have that state / context stored across files. So, Zustand came in handy once again.

**Comment by tcheee** [comment]
> @tcheee @Addminus @chybisov
> 
> Recommendations to improve:
> 
> * improve texts / labels
> * Slider ? (to have that alert flying in and out)
> 
> What do you think?

Awesome. 
Can we do something for the padding/margin? To make it closer to the widget? It seems too far on the screenshots?

**Comment by chybisov** [comment]
@dennyscode, is there any chance to make this as a feature card with a possibility to close, etc.? 

image

File: src/i18n/en/translation.json (hunk 5 above)
Comment (chybisov) [reply src/i18n/en/translation.json:64]: Yes, definitely need to change wording

File: src/components/Widgets/Widgets.tsx [hunk 9]
```diff
@@ -1,15 +1,22 @@
...
+  useSettingsStore,
+} from 'src/stores';
 import type { StarterVariantType } from 'src/types';
 import { WidgetEvents } from './WidgetEvents';
 import { WidgetContainer } from './Widgets.style';
 
 export function Widgets() {
   const { activeTab, setActiveTab } = useActiveTabStore();
+  const { sourceChainToken, destinationChainToken } =
+    useChainTokenSelectionStore();
```
Comment (chybisov) [inline src/components/Widgets/Widgets.tsx:19]: Does this part re-render the widget in response to the event? Please see here https://github.com/lifinance/jumper.exchange/pull/791 why it is bad and restructure the logic if that's needed 🙂

Comment (dennyscode) [reply src/components/Widgets/Widgets.tsx]: THANK YOU! Good point: Check latest commit [refactor: prevent widget re-render](https://github.com/lifinance/jumper.exchange/pull/807/commits/e5d26e9755f697da2ecd890876e44acbc9dee0e8)

**Review by chybisov — APPROVED** [review]

File: src/components/SolanaAlert/SolanaAlert.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
+import { ChainId } from '@lifi/types';
```
Comment (tcheee) [inline src/components/SolanaAlert/SolanaAlert.tsx]: @dennyscode can you rename differently than "SolanaAlert" as it is not a quickfix anymore, please call it "SystemAlert" or "InformationAlert"

**Review by tcheee — APPROVED** [review]
Awesome, lgtm

**Pull request merged**